src/entity/entity.rs

```diff
@@ -2,24 +2,29 @@ use std::time::Duration;
 use std::sync::*;
 
 use crate::gamestate::*;
-use crate::net::pkt::PktPayload;
+
+#[derive(Clone, Copy)]
+pub enum Etype {
+    Player,
+    Enemy
+}
  
 pub trait Entity {
    fn pos(&self) -> &(isize, isize);
    fn mut_pos(&mut self) -> &mut (isize, isize);
    fn move_timeout(&self) -> Duration;
    fn mut_mov_next(&mut self) -> &mut Duration;
-   fn rectify_dir(&self, _gamedata: &Arc<Gamedata>, _pid: usize, dir: (isize, isize)) -> Option<(isize, isize)> {
+   fn rectify_dir(&self, _gamedata: &Arc<Gamedata>, _entid: (Etype, usize), dir: (isize, isize)) -> Option<(isize, isize)> {
        Some(dir)
    }
-   fn on_mov(&mut self, _gamedata: &Arc<Gamedata>, _pid: usize, _prevpos: (isize, isize)) {
+   fn on_mov(&mut self, _gamedata: &Arc<Gamedata>, _entid: (Etype, usize), _prevpos: (isize, isize)) {
    }
-   fn mov(&mut self, gamedata: &Arc<Gamedata>, pid: usize, dir: (isize, isize), now: Duration) -> Option<PktPayload> {
+   fn mov(&mut self, gamedata: &Arc<Gamedata>, entid: (Etype, usize), dir: (isize, isize), now: Duration) -> Option<(isize, isize)> {
        let offcd = *self.mut_mov_next() < now;
        if !offcd {
            return None;
        }
-       let rectified_dir = if let Some(newdir) = self.rectify_dir(gamedata, pid, dir) {
+       let rectified_dir = if let Some(newdir) = self.rectify_dir(gamedata, entid, dir) {
```

Comment:
two assigns here seems redundant u could just use newdir? like declare it b4 if there are scoping issues

Reply:
nice epic bro